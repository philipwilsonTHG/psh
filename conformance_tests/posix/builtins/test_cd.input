# Test cd builtin functionality
mkdir -p /tmp/psh_cd_test/dir1
mkdir -p /tmp/psh_cd_test/dir2

# Test basic cd and pwd
cd /tmp/psh_cd_test/dir1
pwd

# Test cd to relative directory
cd ../dir2
pwd

# Test cd back to dir1
cd ../dir1
pwd

# Test OLDPWD after directory changes
cd ../dir2
echo $OLDPWD

# Test cd - (previous directory)
cd -
pwd

# Test cd with no arguments goes to HOME
cd
echo "Home directory test: $?"

# Test error case - non-existent directory
cd /nonexistent/path 2>/dev/null
echo "Non-existent directory exit status: $?"

# Cleanup
rm -rf /tmp/psh_cd_test