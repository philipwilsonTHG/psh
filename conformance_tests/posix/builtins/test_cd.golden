/private/tmp/psh_cd_test/dir1
/private/tmp/psh_cd_test/dir2
/private/tmp/psh_cd_test/dir1
/private/tmp/psh_cd_test/dir1
/private/tmp/psh_cd_test/dir1
/private/tmp/psh_cd_test/dir1
Home directory test: 0
Non-existent directory exit status: 1
